src/AzureClient/Visualization/HistogramEncoders.cs

```diff
@@ -23,16 +23,17 @@ internal static Histogram ToHistogram(this Stream stream)
         {
             var output = new StreamReader(stream).ReadToEnd();
             var deserializedOutput = JsonConvert.DeserializeObject<Dictionary<string, JToken>>(output);
-            var deserializedHistogram = deserializedOutput["Histogram"] as JArray;
 
             var histogram = new Histogram();
-            for (var i = 0; i < deserializedHistogram.Count - 1; i += 2)
+            if (deserializedOutput["Histogram"] is JArray deserializedHistogram)
             {
-                var key = deserializedHistogram[i].ToObject<string>();
-                var value = deserializedHistogram[i + 1].ToObject<double>();
-                histogram[key] = value;
+                for (var i = 0; i < deserializedHistogram.Count - 1; i += 2)
```

Comment:
Should we assert or validate that the de-serialized histogram has an even number of entries representing key/value pairs, or is that a given that doesn't need validation?

Reply:
Yes, that's a good point. We should at least log a warning so that we detect this case, if it ever happens.